src/replication/ng/adapter/harbor/adapter.go

```diff
@@ -178,6 +179,51 @@ func (a *adapter) GetNamespace(namespace string) (*model.Namespace, error) {
 	}, nil
 }
 
+func (a *adapter) HealthCheck() (model.HealthStatus, error) {
```

Comment:
Put the method into the `DefaultImageRegistry` so that other registries can use it either.